File: adapter/c-sharp/CustomOperation.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Adapter
{
    class CustomOperation
    {
        private IJsonSerializer _jsonSerializer;

        public CustomOperation(IJsonSerializer jsonSerializer)
        {
            _jsonSerializer = jsonSerializer;
        }

        public string SerializeObject(object obj)
        {
            return _jsonSerializer.SerializeObject(obj);
        }
    }
}

File: adapter/php/CustomOperation.php
<?php

namespace adapter\php;

class CustomOperation
{
    private IJsonSerializer $_jsonSerializer;

    public function __construct(IJsonSerializer $jsonSerializer)
    {
        $this->_jsonSerializer = $jsonSerializer;
    }

    public function serializeObject(object $obj): string
    {
        return $this->_jsonSerializer->serializeObject($obj);
    }
}
